tooling/autodispose-rx-error-prone-checker/src/main/java/com/uber/autodispose/rx/error/prone/checker/RxJavaMissingAutoDisposeErrorChecker.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2017. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.rx.error.prone.checker;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.ErrorProneFlags;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Type;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.errorprone.BugPattern.Category.JDK;
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+import static com.google.errorprone.matchers.Matchers.instanceMethod;
+
+/**
+ * Checker for subscriptions not binding to lifecycle in components with lifecycle.
+ * Use -XepOpt:AutoDisposeLeakCheck to add support for custom components with lifecycle.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    name = "RxJavaMissingAutoDisposeErrorChecker",
+    summary = "Always apply an Autodispose scope before subscribing",
+    category = JDK,
+    severity = ERROR
+)
+public final class RxJavaMissingAutoDisposeErrorChecker extends BugChecker
+    implements MethodInvocationTreeMatcher {
+
+  private static final String AS = "as";
+  private static final ImmutableList<MethodMatchers.MethodNameMatcher> AS_CALL_MATCHERS;
+  private static final ImmutableList<MethodMatchers.MethodNameMatcher> TO_CALL_MATCHERS;
+  private static final ImmutableList<MethodMatchers.MethodNameMatcher> SUBSCRIBE_MATCHERS;
+  private static final String SUBSCRIBE = "subscribe";
+  private static final String TO = "to";
```

Comment:
We don't need to support the `to` implementation since it's going away

Reply:
https://github.com/uber/AutoDispose/pull/156/commits/1b21a9eaef625161af97e5c2b5cdecfc6605df9b